libs/blocks/text/text.js

```diff
@@ -52,6 +52,32 @@ function decorateBlockIconArea(el) {
   });
 }
 
+function decorateLinkFarms(el) {
+  loadStyle('/libs/blocks/text/link-farms.css');
+  const title = el.querySelectorAll('.foreground')[0];
+  const foregroundDiv = el.querySelectorAll('.foreground')[1];
+  const count = foregroundDiv.querySelectorAll('h1, h2, h3, h4, h5, h6').length;
+  title.querySelector('h1, h2, h3, h4, h5, h6').classList.add('heading-l');
+  foregroundDiv.querySelectorAll('p').forEach((p) => p.classList.add('body-s'));
+  foregroundDiv.querySelectorAll('div').forEach((divElem, index) => {
+    const heading = divElem.querySelector('h1, h2, h3, h4, h5, h6');
+    heading?.classList.add('heading-xs');
+    if (count) {
+      if (heading) {
```

Comment:
early return with !heading and !count
if (!count) return
if (!heading) divElem.prepend(createTag('h3', { class: 'no-heading heading-xs' }));
...rest of the processing